volume/dir.go

```diff
@@ -44,3 +45,27 @@ func (v *dir) ChildSchemas() []plugin.EntrySchema {
 func (v *dir) List(ctx context.Context) ([]plugin.Entry, error) {
```

Comment:
If we add a `invokeVolumeCB` and `parent` fields to `dir`, then could `volume.List` be moved to `dir.List` ? `dir.List` can do something like
```
if invokeVolumeCB
   // Use itself as the cache key
else
  // Use parent as cache key
end
```
That would also clean-up `volume.List`'s interface to `volume.List(ctx, impl)` so that clients don't have to remember to pass-in an empty path. It would also eliminate the `unDir` stuff.

Reply:
Done.